### PR TITLE
Update for mutation burden

### PR DESCRIPTION
**Pull request by gwaygenomics**
I can no longer use the processed mutation-load data that was provided - on the last phone call, it was recommended that this data should be revisited. Rather than needing to rerun the pipeline once this happens, I decided to scrap this dependency and will make my own mutation burden metric.

The pull request addresses this issue by first generating this data and then updating all future scripts to incorporate it. This was an unexpected issue, but I think the classifier is improved because of it.

### CODE REVIEW

File: scripts/apply_weights.py [hunk 1]
```diff
@@ -136,9 +135,9 @@
 
 # Add information (hypermutated samples and if they were used to train model)
 final_pred = final_pred.assign(hypermutated=1)
-final_pred.loc[final_pred['Silent per Mb'] < 2 *
-               final_pred['Silent per Mb'].std(), 'hypermutated'] = 0
+final_pred.loc[final_pred['log10_mut'] < 5 * final_pred['log10_mut'].std(),
```
Comment (tj8901nm) [inline scripts/apply_weights.py]: Why 5 here? Should it be assigned to a variable?

File: scripts/ddr_summary_figures.R [hunk 2]
```diff
@@ -75,50 +75,51 @@ p <- ggplot(coef_df, aes(x = 1:nrow(coef_df), y = weight)) +
...
+                     offset = c(50, 0, -900, 0))
 ggsave(file = file.path(results_folder, "figures", "ddr_coefficient_plot.pdf"),
        plot = p, height = 2.5, width = 2.75, dpi = 600)
 
 # 3) Plot distributions of predictions according to variant classification
 mut_df <- readr::read_tsv(file.path(results_folder, "tables",
                                     "mutation_classification_scores.tsv"))
-mut_df <- mut_df[, -1]
-colnames(mut_df)[2] <- "index"
+#colnames(mut_df)[2] <- "index"
```
Comment (tj8901nm) [inline scripts/ddr_summary_figures.R]: Is this line needed?

File: scripts/initialize/process_sample_freeze.py [hunk 3]
```diff
@@ -91,14 +94,21 @@
...
+# This command will introduce NAs for these 12 samples, fill them with zeros
+mut_pivot = mut_pivot.loc[freeze_barcodes, :]
+mut_pivot = mut_pivot.fillna(0)
 mut_pivot = mut_pivot.astype(int)
 mut_pivot.to_csv(mut_out_file, sep='\t', index_col=0)
 
+# Generate a mutation burden variable (log10 total deleterious mutations)
+burden_df = mutation_df[mutation_df['Variant_Classification'].isin(mutations)]
+burden_df = burden_df.groupby('SAMPLE_BARCODE').apply(len)
+burden_df = np.log(burden_df)
```
Comment (tj8901nm) [inline scripts/initialize/process_sample_freeze.py]: I think `np.log` is natural logarithm. Perhaps `np.log10` here?

File: scripts/ddr_summary_figures.R (hunk 2 above)
Comment (gwaygenomics) [inline scripts/ddr_summary_figures.R]: nope, forgot to remove

File: scripts/initialize/process_sample_freeze.py (hunk 3 above)
Comment (gwaygenomics) [inline scripts/initialize/process_sample_freeze.py]: great catch! thanks @tj8901nm !

File: scripts/apply_weights.py (hunk 1 above)
Comment (gwaygenomics) [inline scripts/apply_weights.py]: i think i will leave this hardcoded in to ensure consistency. The goal of this line is to remove the hypermutated outliers if the user thinks that adjusting for them in the model is not sufficient to control for their effect. 5 std ends up removing about the top 300 samples if specified

**Comment by gwaygenomics** [comment]
ok @tj8901nm - ready for your review again. Thanks! (only one more PR to go I think)

**Pull request merged**